Add screenshot feature

Kirpal opened the pull request:
Added the ability to take a screenshot of the electron window, using the shortcut <kbd>CmdOrCtrl</kbd> <kbd>Shift</kbd> <kbd>S</kbd>

This is just the shortcut I chose, it's totally up for debate. I also incremented the version to 1.2.0.

parro-it commented:
Seems a useful feauture @Kirpal :+1: I will use it for sure!

It would be also useful to have an option or another shortcut to copy the screenshot to clipboard instead of saving it to file. This simplify the use case of taking a screenshot to include it in a GH issue or other similar target...

Regarding the shortcut, `CmdOrCtrl Shift S` seems more appropriate for a `Save`  or `Save as` action.
What about `PrintScreen` ?

Kirpal commented:
@parro-it that is a good point! I think having a second shortcut for saving to clipboard would be a better choice than having it set as an option.

That is also a good point about the shortcut, but I think `PrintScreen` might also not be the best idea, because it could interfere with the system screenshot shortcut...

Kirpal commented:
@parro-it I added saving image to clipboard, through a message box that pops up asking what the user would like to do with the image.

I changed the shortcuts to:

- macOS: <kbd>Cmd</kbd> <kbd>Alt</kbd> <kbd>3</kbd>
- Linux: <kbd>Ctrl</kbd> <kbd>PrintScreen</kbd>
- Windows: <kbd>Ctrl</kbd> <kbd>PrintScreen</kbd>

sindresorhus commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "electron-debug",
-  "version": "1.1.0",
+  "version": "1.2.0",
```
Don't bump version on PRs. It's up to the maintainer to choose when and what version to use.

sindresorhus commented:
> I think having a second shortcut for saving to clipboard would be a better choice than having it set as an option.

👍

sindresorhus commented:
I think it should just save the screenshot on the Desktop instead of opening a save dialog. That's what the native screenshot feature on macOS does. When I capture screenshots I don't want to be bothered to choose where to save.

Kirpal commented:
@sindresorhus Sorry for responding so late... I changed it so it saves to the desktop instead of opening a dialog. Do you think there should be separate shortcuts for clipboard/desktop, or a dialog box to choose (like I have right now).

Kirpal commented:
@sindresorhus What if it was like on ChromeOS, where it saves it automatically to a folder, and then has a popup with a copy to clipboard button?

sindresorhus commented:
> Do you think there should be separate shortcuts for clipboard/desktop, or a dialog box to choose (like I have right now).

Separate shortcuts.

> What if it was like on ChromeOS, where it saves it automatically to a folder, and then has a popup with a copy to clipboard button?

That would be annoying. In most cases I just want it on the Desktop. If I want to copy, I can use the other keyboard shortcut. Modal prompt dialogs are IMHO bad UX.

sindresorhus commented:
@Kirpal Still interested in finishing this? :)

Kirpal commented:
@sindresorhus yep... I'll get around to it eventually, I've just been very busy with school. I haven't had time to code in weeks it seems like :(

sindresorhus commented:
Closing this to clean up my PR queue. Let me know if you ever get to this and I'll reopen it :)
